core/client/fs/src/main/java/alluxio/client/block/stream/LocalFileDataWriter.java

```diff
@@ -135,9 +134,6 @@ public void writeChunk(final ByteBuf buf) throws IOException {
 
   @Override
   public void cancel() throws IOException {
-    if (mStream.isClosed() || mStream.isCanceled()) {
-      return;
-    }
```

Comment:
Should we do a similar pattern here and register `mStream.cancel()` with the closer and just close the closer?

Reply:
sgtm

Reply:
@calvinjia done